src/find_libpython/__init__.py

```diff
@@ -130,6 +130,19 @@ def wrapper(*args, **kwds):
 
     return wrapper
 
+def _get_proc_library():
+    pid = os.getpid()
+    path = f"/proc/{pid}/maps"
+    _logger.debug(f"Checking {path} for libpython")
+    lines = open(path).readlines()
+
+    for l in lines:
+        path = l.split(" ")[0]
+        if "libpython" in os.path.basename(path):
+            if not os.path.isfile(path): continue
+            return path
+
```

Comment:
There's a case where nothing matching `libpython` is found. This function needs to be able to yield nothing. It's best to turn it into a chained generated. `yield` in this function instead of `return` and `yield from` in the call to this function.

Reply:
Sure. thanks